src/main/java/org/zalando/nakadi/config/SecuritySettings.java

```diff
@@ -12,6 +12,7 @@
     public enum AuthMode {
         OFF,   // no authentication at all
         BASIC, // only checks that the token is valid (has "uid" scope)
+        REALM, // checks that the token is valid and contains at least one required realm
```

Comment:
Why have you decided to make it a 4th auth mode and not just additional pluggable auth. feature?